src/functionality/configUtils.ts

```diff
@@ -91,3 +113,247 @@ export function handleCustomCSS(styleSheetId: string, styles: string): void {
   stylesheet.appendChild(styleTextNode);
   document.head.appendChild(stylesheet);
 }
+
+//-----------------------------------
+// handleExtentSelector
+// ---------------------------------
+
+/**
+ * Handles updating the view's constraint without filtering out features.
+ * Watch for changes in mapArea, extentSelector, extentSelectorConfig.
+ */
+export function handleExtentSelector(
+  config: any,
+  view: __esri.MapView | __esri.SceneView
+) {
+  if (view?.type === "2d") {
+    const map = view.map as __esri.WebMap;
+    const { initialViewProperties } = map;
+    const { mapArea, extentSelector, extentSelectorConfig } = config;
+    const isMapAreaOn = mapArea == null ? true : mapArea;
+    if (isMapAreaOn && extentSelector && extentSelectorConfig) {
+      const { constraints, mapRotation } = extentSelectorConfig;
+      const tempConstraints: __esri.MapViewConstraints = {
+        ...constraints,
+        geometry: fromJSON(constraints?.geometry)?.extent
+      };
+      tempConstraints.minScale = +constraints.minScale;
+      tempConstraints.maxScale = +constraints.maxScale;
+      view.constraints = tempConstraints;
+      setMapRotation(view, mapRotation);
+    } else {
+      view.rotation = initialViewProperties.viewpoint?.rotation ?? 0;
+      view.constraints.geometry = undefined;
+      view.constraints.minZoom = -1;
+      view.constraints.maxZoom = -1;
+      view.constraints.minScale = 0;
+      view.constraints.maxScale = 0;
+      view.constraints.rotationEnabled = true;
+    }
+  }
+}
+
+/**
+ *  Handles updating the view's constraint and uses extent to filter out features.
+ *  Watch for changes in mapArea, extentSelector, extentSelectorConfig
+ */
+function handleExtentSelectorLVFilter(
+  config: any,
+  view: __esri.MapView | __esri.SceneView,
+  layerView?: ExtentLayerView
+) {
+  if (view?.type === "2d") {
+    const filterLayers = [
```

Comment:
this is defined in more than one function with the same value. I think it's safe to define this once outside of the functions